src/main/java/org/opensearch/security/user/User.java

```diff
@@ -305,4 +337,27 @@ public boolean isServiceAccount() {
     public boolean isPluginUser() {
         return name != null && name.startsWith("plugin:");
     }
+
+    public String toSerializedBase64() {
+        String result = this.serializedBase64;
+
+        if (result == null) {
+            this.serializedBase64 = result = Base64JDKHelper.serializeObject(this);
+        }
+
+        return result;
+    }
+
+    /**
+     * Creates a backwards compatible object that can be used for serialization
+     */
+    Object writeReplace() {
+        return new org.opensearch.security.user.serialized.User(this);
+    }
```

Comment:
FYI: This does not work correctly yet. The serialized output will refer to the class `org.opensearch.security.user.serialized.User` which is not known in previous OpenSearch versions. Thus, mixed clusters fail. 

I am not completely sure how to fix this. One option would be to move the "actual" User class to a different package and just keep the dummy for serialization here. But I do not like this approach too much.